core/src/main/scala/neotypes/model.scala

```diff
@@ -0,0 +1,198 @@
+  }
+
+  /** Represents a Neo4j Node. */
+  final case class Node(
+      elementId: String,
+      labels: Set[String],
+      properties: Map[String, Value]
+  ) extends Entity {
+    /** Checks if this Node contains the given label; case insensitive. */
+    def hasLabel(label: String): Boolean =
+      labels.contains(label.trim.toLowerCase)
+  }
+
+  /** Represents a Neo4j Relationship. */
+  final case class Relationship(
+      elementId: String,
+      relationshipType: String,
+      properties: Map[String, Value],
+      startNodeId: String,
+      endNodeId: String
+  ) extends Entity {
+    /** Checks if this Relationship has the given type; case insensitive. */
+    def hasType(tpe: String): Boolean =
+      relationshipType == tpe.trim.toLowerCase
+  }
+
+  /** Represents a Neo4j Path. */
+  sealed trait Path extends NeoType {
+    def start: Node
+    def end: Node
+
+    def nodes: List[Node]
+    def relationships: List[Relationship]
+    def segments: List[Path.Segment]
+
+    def contains(node: Node): Boolean
+    def contains(relationship: Relationship): Boolean
+  }
+  object Path {
+    final case class EmptyPath(node: Node) extends Path {
+      override final val start: Node = node
+      override final val end: Node = node
+      override final val nodes: List[Node] = node :: Nil
+      override final val relationships: List[Relationship] = Nil
+      override final val segments: List[Segment] = Nil
+
+      override def contains(node: Node): Boolean =
+        this.node == node
+
+      override def contains(relationship: Relationship): Boolean =
+        false
+    }
+
+    final case class NonEmptyPath(segments: ::[Segment]) extends Path {
+      override final val start: Node =
+        segments.head.start
+
+      override def end: Node =
+        segments.last.end
+
+      override def nodes: List[Node] =
+        start :: segments.map(s => s.end)
+
+      override def relationships: List[Relationship] =
+        segments.map(s => s.relationship)
+
+      override def contains(node: Node): Boolean =
+        start == node || segments.exists(s => s.end == node)
+
+      override def contains(relationship: Relationship): Boolean =
+        segments.exists(s => s.relationship == relationship)
+    }
+
+    final case class Segment(start: Node, relationship: Relationship, end: Node)
+  }
+
+  /** Parent type of all Neo4j property types. */
+  sealed trait Value extends NeoType
+  object Value {
+    final case class ListValue[V <: SimpleValue](values: Iterable[V]) extends Value
+    sealed trait SimpleValue extends Value
+    sealed trait NumberValue extends SimpleValue
+    final case class Integer(value: Int) extends NumberValue
+    final case class Decimal(value: Double) extends NumberValue
+    final case class Str(value: String) extends SimpleValue
+    final case class Bool(value: Boolean) extends SimpleValue
+    final case class Bytes(value: ArraySeq[Byte]) extends SimpleValue
```

Comment:
scodec-bits `ByteVector` could be good here. The library has no dependencies and is very stable.

Reply:
Ditto dependencies :)